include/battle.h

```diff
@@ -718,20 +719,22 @@ struct BattleStruct
     uq4_12_t supremeOverlordModifier[MAX_BATTLERS_COUNT];
     u8 itemPartyIndex[MAX_BATTLERS_COUNT];
     u8 itemMoveIndex[MAX_BATTLERS_COUNT];
-    bool8 trainerSlideHalfHpMsgDone;
+    u8 trainerSlideHalfHpMsgDone:1;
     u8 trainerSlideFirstCriticalHitMsgState:2;
     u8 trainerSlideFirstSuperEffectiveHitMsgState:2;
     u8 trainerSlideFirstSTABMoveMsgState:2;
     u8 trainerSlidePlayerMonUnaffectedMsgState:2;
-    bool8 trainerSlideMegaEvolutionMsgDone;
-    bool8 trainerSlideZMoveMsgDone;
-    bool8 trainerSlideBeforeFirstTurnMsgDone;
-    bool8 trainerSlideDynamaxMsgDone;
+    u8 trainerSlideMegaEvolutionMsgDone:1;
+    u8 trainerSlideZMoveMsgDone:1;
+    u8 trainerSlideBeforeFirstTurnMsgDone:1;
+    u8 trainerSlideDynamaxMsgDone:1;
```

Comment:
Excellent idea, but `trainerSlidePlayerMonUnaffectedMsgState` currently takes up bits 8 and 9, which means that there's a bit being lost. Feel free to correct me, but maybe a `u16` could be interesting, and possible with padding undearneath `trainerSlideDynamaxMsgDone` to show how many bits are still free for extra trainer slide messages?

Reply:
Could I just move `trainerSlideHalfHpMsgDone` down a bit so that `trainerSlidePlayerMonUnaffectedMsgState` gets bit 7 and 8? I'm not sure about pedding since it isn't added anywhere in the BattleStruct.